packages/api/src/util/middleware.js

```diff
@@ -21,16 +21,18 @@ export const notExactlyOneValue = (values) => {
 };
 
 // Checks an array of object keys against object -- at least one of the properties is defined
-export const oneExists = (keys, object) => {
+export const someExists = (keys, object) => {
   return keys.some((key) => key in object);
 };
 
 // Checks an array for at least one truthy value
-export const oneTruthy = (values) => values.some((value) => !!value);
+export const someTruthy = (values) => values.some((value) => !!value);
 
-// Sets falsy values to null for editing values that may have values for exclusive constraints -- does not handle zeros yet
+// Sets falsy values to null for editing values that may have values for exclusive constraints
 export const setFalsyValuesToNull = (array, obj) => {
   for (const val of array) {
-    obj[val] = obj[val] || null;
+    if (obj[val] !== 0 && obj[val] !== false) {
+      obj[val] ??= null;
```

Comment:
I think you can remove `if (obj[val] !== 0 && obj[val] !== false) {`...

Reply:
Sorry I resolved the other comment I thought I understood, removed the "todo in future" part and just did it now. We just discussed handling these cases: https://github.com/LiteFarmOrg/LiteFarm/pull/3484#discussion_r1805582170

Reply:
Ah sorry, I meant `obj[val] ??= null` should be enough!
```
const obj = { zero: 0, false: false }; 
obj.zero ??=null; // obj.zero is still 0 
obj.false ??=null; // obj.false is still false
```

Reply:
...damn chatGPT lol 🙏  thanks!

Reply:
https://github.com/LiteFarmOrg/LiteFarm/pull/3510/commits/1b353cd8fb5ae8efd1f10e7d5c57dfe3816c7fa5